### PR TITLE
Dynamically add open graph meta tags

### PR DESCRIPTION
**Pull request by samuelli**
Partly addresses #381 

This filters requests from static bot user agents which don't render webpages to the bot-render service, allowing for dynamically generated meta tags.

The tags are added just for `/element/*` pages. Image will use first image in readme file, falling back to the author's avatar.

### CODE REVIEW

File: client/bot-filter.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+class FilterUserAgent(webapp2.RequestHandler):
+  def get(self, path):
+    filtered = r'baiduspider|facebookexternalhit|twitterbot|rogerbot|linkedinbot|embedly|quora\ link\ preview|showyoubot|outbrain|pinterest|slackbot|vkShare|W3C_Validator';
+    filter = re.search(filtered, str(self.request.headers['User-Agent']))
+
+    if filter:
+      try:
+        # Set deadline for fetch to 10 seconds
+        urlfetch.set_default_fetch_deadline(20)
```
Comment (dstockwell) [inline client/bot-filter.py]: 20 means 10?

File: client/bot-filter.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        result = urlfetch.fetch('https://bot-render.appspot.com/?url=%s' % self.request.url)
+        if result.status_code == 200:
+          self.response.write(result.content)
+        else:
+          self.response.status_code = result.status_code
+        return
+      except urlfetch.Error:
+          logging.exception('Caught exception fetching url')
```
Comment (dstockwell) [inline client/bot-filter.py]: indentation?

File: client/bot-filter.py (hunk 1 above)
Comment (samuelli) [inline client/bot-filter.py]: Fixed

File: client/bot-filter.py (hunk 2 above)
Comment (samuelli) [inline client/bot-filter.py]: Fixed

**Pull request merged**